Source/SeaInk.Core/Entities/Tables/ExcelTable.cs

```diff
@@ -0,0 +1,209 @@
+﻿using System.Collections.Generic;
+using System.IO;
+using ClosedXML.Excel;
+using SeaInk.Core.Models.Tables;
+
+namespace SeaInk.Core.Entities.Tables
+{
+    public class ExcelTable : ITable
+    {
+        private XLWorkbook _workbook = null!;
+        private string _pathToFile;
+
+
+        public int SheetCount => _workbook.Worksheets.Count;
+        
+        public int ColumnCount(TableIndex index)
+        {
+            return 0;
+        }
+
+        public int RowCount(TableIndex index)
+        {
+            return 0;
+        }
+
+        public void CreateSheet(TableIndex index)
+        {
+            _workbook.Worksheets.Add(index.SheetName);
+            Save();
+        }
+
+        public void DeleteSheet(TableIndex sheet)
+        {
+            _workbook.Worksheets.Delete(sheet.SheetName);
+
+            var oldPath = _pathToFile;
+            _pathToFile = SeparatePath(_pathToFile).path;
+            _pathToFile += "t.xlsx";
+            
+            Save();
+            File.Delete(oldPath);
+            
+            Rename(SeparatePath(oldPath).name);
+
+        }
+
+        public void Load(string address)
+        {
+            _workbook = new XLWorkbook(address);
+            _pathToFile = address;
+            Save();
+        }
+
+        public string Create(string path)
+        {
+            _workbook = new XLWorkbook();
+            _workbook.AddWorksheet("Important Sheet");
+            
+            _pathToFile = path;
+            return path;
+        }
+
+        private (string path, string name) SeparatePath(string pathToFile)
+        {
+
+            var array = new List<string>();
+            var word = "";
+            for (var i = 0; i < pathToFile.Length; i++)
+            {
+                word += pathToFile[i];
+                if (pathToFile[i] == '\\' || i == pathToFile.Length - 1)
+                {
+                    array.Add(word);
+                    word = "";
+                }
+            }
+            
+            var path = "";
+            for (var i = 0; i < array.Count - 1; ++i)
+            {
+                path += array[i];
+            }
+
+            (string path, string name) t;
+            t.path = path;
+            t.name = array[^1];
+            return t;
+        }
+        
+        public void Rename(string name)
+        {
+            var oldPath = _pathToFile;
+            _pathToFile = SeparatePath(_pathToFile).path;
+            _pathToFile += name;
+            
+            Save();
+            File.Delete(oldPath);
+        }
+
+        public void RenameSheet(TableIndex index, string name)
+        {
+            _workbook.Worksheet(index.SheetName).Name = name;
+            Save();
+        }
+
+        private void Save()
+        {
+            if (File.Exists(_pathToFile))
+            {
+                File.Delete(_pathToFile);
+            }
+
+            _workbook.SaveAs(_pathToFile);
```

Comment:
Я проверила, что `SaveAs` из коробки сам перезаписывает существующий файл новым
```suggestion
            _workbook.SaveAs(_pathToFile);
```